validator/schemas/relationship-schema-v1.json

```diff
@@ -0,0 +1,237 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema",
+  "$id": "relationship-schema-v1",
+  "type": "object",
+  "title": "Entity Relationship synthesis schema",
+  "description": "Schema used to verify that a given entity relationship yaml is valid",
+  "required": [
+    "relationships"
+  ],
+  "additionalProperties": false,
+  "properties": {
+    "relationships": {
+      "$id": "#/relationships",
+      "type": "array",
+      "title": "Rules for Relationship Synthesis",
+      "description": "Information necessary to synthesize the relatioship between two entity types",
+      "minItems": 1,
+      "examples": [{
+        "instrumentation.provider": "newrelic",
+        "eventType": ["Span"],
+        "relationship": "CALLS",
+        "target": {
+          "domain": "EXT",
+          "type": "SERVICE",
+          "domainId": "peer.hostname",
+          "encodeIdentifierInGUID": true,
+          "processor": "hostExtractor"
+        },
+        "conditions": [{
+          "attribute": "requestUrl"
+        }]
+      }],
+      "items": {
+        "$id": "#/relationships/items",
+        "type": "object",
+        "oneOf" : [
+          {"required" : [
+            "instrumentation.provider",
+            "eventType",
+            "relationship",
+            "source"
+          ]},
+          {"required" : [
+            "instrumentation.provider",
+            "eventType",
+            "relationship",
+            "target"
+          ]}
+        ],
+        "additionalProperties": false,
+        "properties": {
+          "instrumentation.provider": {
+            "$id": "#relationships/items/instrumentation.provider",
+            "type": "string",
+            "title": "Instrumentation Provider",
+            "description": "instrumention.provider agent or source from where the telemetry is coming",
+            "examples": ["newrelic"]
+          },
+          "eventType" : {
```

Comment:
wondering if this should be optional or not. Like synthesis definition?

Reply:
What would the default be?